Document Clique transitions (#1583)
hyperledger/besu-docs: docs/private-networks/how-to/configure/consensus/clique.md

Define the number of blocks between epoch transitions in the [Clique genesis file](#genesis-file).

## Transitions

The `transitions` genesis configuration item allows you to specify a future block number at which to
change the Clique network configuration in an existing network.
For example, you can update the [block time](#configure-block-time-on-an-existing-network) and

> whether to [create empty blocks](#configure-empty-blocks-on-an-existing-network).

Reviewer: ```suggestion
choose whether to [create empty blocks](#configure-empty-blocks-on-an-existing-network).
```

Author: The meaning is "you can update the block time, and _you can update_ whether to create empty blocks..." so adding "choose" would change the meaning.

    - `<NewValue>` is the updated value for `blockperiodseconds`.

    <Tabs>
    <TabItem value="Syntax" label="Syntax" default>

    ```json
    {
      "config": {
        ...
        "clique": {
          "blockperiodseconds": 3,
          "epochlength": 30,
          "requesttimeoutseconds": 6,
          "createemptyblocks": true
        },
        "transitions": {
          "clique": [
            {
              "block": <FutureBlockNumber>,
              "blockperiodseconds": <NewValue>
            }
          ]
        }
      },
      ...
    }
    ```

    </TabItem>
    <TabItem value="Example" label="Example">

    ```json
    {
      "config": {
        ...
        "clique": {
          "blockperiodseconds": 3,
          "epochlength": 30,
          "requesttimeoutseconds": 6,
          "createemptyblocks": true
        },
        "transitions": {
          "clique": [
            {
              "block": 3,
              "blockperiodseconds": 1
            },
            {
              "block": 6,
              "blockperiodseconds": 2
            },
          ]
        }
      },
      ...
    }
    ```

    </TabItem>
    </Tabs>

> 3. Restart all nodes in the network using the updated genesis file.

Reviewer: ```suggestion
1. Restart all nodes in the network using the updated genesis file.
```

> 4. To verify the changes after the transition block, view the Besu logs and check that the time

Reviewer: ```suggestion
1. To verify the changes after the transition block, view the Besu logs and check that the time
```

   difference between each block matches the updated block period.

### Configure empty blocks on an existing network

To update an existing network with a new [`createemptyblocks`](#skip-empty-blocks):

1. Stop all nodes in the network.

> 2. In the [genesis file](#genesis-file), add the `transitions` configuration item where:

Reviewer: ```suggestion
1. In the [genesis file](#genesis-file), add the `transitions` configuration where:
```

        "clique": {
          "blockperiodseconds": 3,
          "epochlength": 30,
          "requesttimeoutseconds": 6,
          "createemptyblocks": true
        },
        "transitions": {
          "clique": [
            {
              "block": <FutureBlockNumber>,
              "blockperiodseconds": <NewValue>
            }
          ]
        }
      },
      ...
    }
    ```

    </TabItem>
    <TabItem value="Example" label="Example">

    ```bash
    {
      "config": {
        ...
        "clique": {
          "blockperiodseconds": 3,
          "epochlength": 30,
          "requesttimeoutseconds": 6,
          "createemptyblocks": true
        },
        "transitions": {
          "clique": [
            {
              "block": 3,
              "blockperiodseconds": 1
            },
            {
              "block": 6,
              "blockperiodseconds": 2
            },
          ]
        }
      },
      ...
    }
    ```

    </TabItem>
    </Tabs>

3. Restart all nodes in the network using the updated genesis file.
4. To verify the changes after the transition block, call
   [`clique_getSigners`](../../../reference/api/index.md#clique_getsigners),

>    specifying `latest`.

Reviewer 2: This RPC won't help verify the block period has correctly changed. If you look at the output of that RPC, it doesn't make sense as there's nothing about block times in it...
https://besu.hyperledger.org/private-networks/reference/api#clique_getsigners
```
{
    "jsonrpc" : "2.0",
    "id" : 1,
    "result" : [ "0x42eb768f2244c8811c63729a21a3569731535f06", "0x7ffc57839b00206d1ad20c69a1981b489f772031", "0xb279182d99e65703f0076e4812653aab85fca0f0" ]
}
```

I think the docs for QBFT and IBFT are wrong on this point too, probably a copy-paste error from another transition.

For this one, I would verify it by looking at the logs since you can see the timestamp for each produced/imported block and calculate the time difference between them (i.e. block period).

Reviewer 2: Must have been copied from https://besu.hyperledger.org/private-networks/how-to/configure/consensus/add-validators-without-voting before that was its own page
